dipy/tracking/local/tissue_classifier.pyx

```diff
@@ -1,13 +1,55 @@
 cimport cython
 cimport numpy as np
 
-from .interpolation cimport trilinear_interpolate4d, _trilinear_interpolate_c_4d
+from libc.math cimport round
+
+from .interpolation cimport(trilinear_interpolate4d,
+                            _trilinear_interpolate_c_4d)
 
 cdef class TissueClassifier:
     cpdef TissueClass check_point(self, double[::1] point):
         pass
 
 
+cdef class BinaryTissueClassifier(TissueClassifier):
+    """
+    cdef:
+        unsigned char[:, :, :] mask
+    """
+
+    def __cinit__(self, mask):
+        self.interp_out_view = self.interp_out_double
+        self.mask = (mask > 0).astype('uint8')
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cpdef TissueClass check_point(self, double[::1] point):
+        cdef:
+            double result
+            int err
+            int voxel[3]
+
+        if point.shape[0] != 3:
+            raise ValueError("Point has wrong shape")
+
+        voxel[0] = int(round(point[0]))
+        voxel[1] = int(round(point[1]))
+        voxel[2] = int(round(point[2]))
+
+        if (voxel[0] < 0 or voxel[0] > self.mask.shape[0]
+                or voxel[1] < 0 or voxel[1] > self.mask.shape[1]
+                or voxel[2] < 0 or voxel[2] > self.mask.shape[2]):
+            return OUTSIDEIMAGE
+
+        result = self.mask[voxel[0], voxel[1], voxel[2]]
```

Comment:
I changed the result type to unsigned char, as it should be. 

I was wondering why having the type set to 'double' and assigning an 'unsigned char' did not make it crash? I guess it worked cause I'm just checking for 0 or >0.